Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -489,93 +504,133 @@ void CGContextAddRects(CGContextRef context, const CGRect* rects, unsigned count
 */
 void CGContextAddLineToPoint(CGContextRef context, CGFloat x, CGFloat y) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+
+    CGPathAddLineToPoint(context->Path(), &context->CurrentGState().transform, x, y);
 }
 
 /**
  @Status Interoperable
 */
 void CGContextAddCurveToPoint(CGContextRef context, CGFloat cp1x, CGFloat cp1y, CGFloat cp2x, CGFloat cp2y, CGFloat x, CGFloat y) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+
+    CGPathAddCurveToPoint(context->Path(), &context->CurrentGState().transform, cp1x, cp1y, cp2x, cp2y, x, y);
 }
 
 /**
  @Status Interoperable
 */
 void CGContextAddQuadCurveToPoint(CGContextRef context, CGFloat cpx, CGFloat cpy, CGFloat x, CGFloat y) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+
+    CGPathAddQuadCurveToPoint(context->Path(), &context->CurrentGState().transform, cpx, cpy, x, y);
 }
 
 /**
  @Status Interoperable
 */
 void CGContextMoveToPoint(CGContextRef context, CGFloat x, CGFloat y) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+
+    CGPathMoveToPoint(context->Path(), &context->CurrentGState().transform, x, y);
 }
 
 /**
  @Status Interoperable
 */
 void CGContextAddArc(CGContextRef context, CGFloat x, CGFloat y, CGFloat radius, CGFloat startAngle, CGFloat endAngle, int clockwise) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+
+    CGPathAddArc(context->Path(), &context->CurrentGState().transform, x, y, radius, startAngle, endAngle, clockwise);
 }
 
 /**
  @Status Interoperable
 */
 void CGContextAddArcToPoint(CGContextRef context, CGFloat x1, CGFloat y1, CGFloat x2, CGFloat y2, CGFloat radius) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+
+    CGPathAddArcToPoint(context->Path(), &context->CurrentGState().transform, x1, y1, x2, y2, radius);
 }
 
 /**
  @Status Interoperable
 */
 void CGContextAddEllipseInRect(CGContextRef context, CGRect rect) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+
+    CGPathAddEllipseInRect(context->Path(), &context->CurrentGState().transform, rect);
 }
 
 /**
  @Status Interoperable
 */
 void CGContextAddPath(CGContextRef context, CGPathRef path) {
     NOISY_RETURN_IF_NULL(context);
-    // The Apple SDK docs imply that passing a NULL path is valid.
-    // So avoid calling into the backing if NULL.
-    if (path) {
-        UNIMPLEMENTED();
+    if (!path) {
+        return;
+    }
+
+    if (!context->HasPath()) {
+        // If we don't curerntly have a path, take this one in as our own.
+        woc::unique_cf<CGMutablePathRef>& currentPathRef{context->Impl().currentPath};
+        currentPathRef.reset(CGPathCreateMutableCopy(path));
+        return;
```

Comment:
no.

---
In reply to: [85366284](https://github.com/Microsoft/WinObjC/pull/1240#discussion_r85366284) [](ancestors = 85366284)